apps/vaults/contexts/useActionFlow.tsx

```diff
@@ -0,0 +1,226 @@
+import {ETH_TOKEN_ADDRESS, WETH_TOKEN_ADDRESS, WFTM_TOKEN_ADDRESS} from '@yearn-finance/web-lib/utils/constants';
+import {formatBN, formatToNormalizedValue} from '@yearn-finance/web-lib/utils/format.bigNumber';
+import performBatchedUpdates from '@yearn-finance/web-lib/utils/performBatchedUpdates';
+import {useWallet} from '@common/contexts/useWallet';
+import {DefaultTNormalizedBN} from '@common/utils';
+
+import {Solvers} from './useSolver';
+import {useWalletForZap} from './useWalletForZaps';
+
+import type {ReactNode} from 'react';
+import type {TDropdownOption, TNormalizedBN} from '@common/types/types';
+import type {TYearnVault} from '@common/types/yearn';
+
+type	TActionFlowContext = {
+	currentVault: TYearnVault;
+	possibleOptionsFrom: TDropdownOption[];
+	possibleOptionsTo: TDropdownOption[];
+	selectedOptionFrom: TDropdownOption | undefined;
+	selectedOptionTo: TDropdownOption | undefined;
+	amount: TNormalizedBN;
+	onChangeAmount: (amount: TNormalizedBN) => void;
+	onUpdateSelectedOptionFrom: (option: TDropdownOption) => void;
+	onUpdateSelectedOptionTo: (option: TDropdownOption) => void;
+	onSwitchSelectedOptions: () => void;
+	isDepositing: boolean;
+	maxDepositPossible: TNormalizedBN;
+	currentSolver: Solvers;
+}
+const	DefaultActionFlowContext: TActionFlowContext = {
+	currentVault: {} as TYearnVault, // eslint-disable-line @typescript-eslint/consistent-type-assertions
+	possibleOptionsFrom: [],
+	possibleOptionsTo: [],
+	selectedOptionFrom: undefined,
+	selectedOptionTo: undefined,
+	amount: DefaultTNormalizedBN,
+	onChangeAmount: (): void => undefined,
+	onUpdateSelectedOptionFrom: (): void => undefined,
+	onUpdateSelectedOptionTo: (): void => undefined,
+	onSwitchSelectedOptions: (): void => undefined,
+	isDepositing: true,
+	maxDepositPossible: DefaultTNormalizedBN,
+	currentSolver: Solvers.VANILLA
+};
+
+const ActionFlowContext = createContext<TActionFlowContext>(DefaultActionFlowContext);
+function ActionFlowContextApp({children, currentVault}: {children: ReactNode, currentVault: TYearnVault}): React.ReactElement {
+	const {balances} = useWallet();
+	const {safeChainID} = useChainID();
+	const {networks} = useSettings();
+	const {balances: zapBalances, tokensList} = useWalletForZap();
+
+	const [possibleOptionsFrom, set_possibleOptionsFrom] = useState<TDropdownOption[]>([]);
+	const [possibleZapOptionsFrom, set_possibleZapOptionsFrom] = useState<TDropdownOption[]>([]);
+	const [possibleOptionsTo, set_possibleOptionsTo] = useState<TDropdownOption[]>([]);
+	const [selectedOptionFrom, set_selectedOptionFrom] = useState<TDropdownOption | undefined>();
+	const [selectedOptionTo, set_selectedOptionTo] = useState<TDropdownOption | undefined>();
+	const [amount, set_amount] = useState<TNormalizedBN>(DefaultTNormalizedBN);
+
+	const isDepositing = useMemo((): boolean => (
+		!selectedOptionTo?.value ? true : toAddress(selectedOptionTo.value) === toAddress(currentVault.address)
```

Comment:
Can we simplify this to
```suggestion
		!selectedOptionTo?.value || toAddress(selectedOptionTo.value) === toAddress(currentVault.address)
```